ocaml-variants.4.11.1+BER: there are errors in configure's option

Although `./configure -prefix %{prefix}% -cc cc -aspp cc -c` is used for compiling OCaml in some packages (e.g. [ocaml-variants.4.07.1+BER](https://github.com/ocaml/opam-repository/blob/a745d1a17e803a58672c2c239e3821e3f7c4bc36/packages/ocaml-variants/ocaml-variants.4.07.1%2BBER/opam)), it doesn't seems to work for OCaml 4.11.1.

The following is the result of `opam switch create 4.11.1+BER`.
```
<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫 
[ocaml-variants.4.11.1+BER] downloaded from cache at https://opam.ocaml.org/cache

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫 
∗ installed base-bigarray.base
∗ installed base-threads.base
∗ installed base-unix.base
[ERROR] The compilation of ocaml-variants failed at "/Users/moatom/.opam/opam-init/hooks/sandbox.sh build ./configure -prefix
        /Users/moatom/.opam/4.11.1+BER -cc cc -aspp cc -c".

#=== ERROR while compiling ocaml-variants.4.11.1+BER ==========================#
# context     2.0.7 | macos/x86_64 |  | https://opam.ocaml.org#dddf7f21
# path        ~/.opam/4.11.1+BER/.opam-switch/build/ocaml-variants.4.11.1+BER
# command     ~/.opam/opam-init/hooks/sandbox.sh build ./configure -prefix /Users/moatom/.opam/4.11.1+BER -cc cc -aspp cc -c
# exit-code   1
# env-file    ~/.opam/log/ocaml-variants-2623-d8a4ae.env
# output-file ~/.opam/log/ocaml-variants-2623-d8a4ae.out
### output ###
# configure: error: unrecognized option: `-cc'
# Try `./configure --help' for more information



<><> Error report <><><><><><><><><><><><><><><><><><><><><><><><><><><><><>  🐫 
┌─ The following actions failed
│ λ build ocaml-variants 4.11.1+BER
└─ 
┌─ The following changes have been performed (the rest was aborted)
│ ∗ install base-bigarray base
│ ∗ install base-threads  base
│ ∗ install base-unix     base
└─ 
```
I haven't gotten this type of error in installing 4.04.0+BER, 4.02.1+BER, 4.07.1+BER.

I could personally avoid this error by changing `["./configure" "-prefix" "%{prefix}%" "-cc" "cc" "-aspp" "cc -c"]` in https://github.com/ocaml/opam-repository/pull/17365 to ` ["./configure" "--prefix=%{prefix}%" "CC=cc" "ASPP=cc -c"] `.

--- a/packages/ocaml-variants/ocaml-variants.4.11.1+BER/opam
+++ b/packages/ocaml-variants/ocaml-variants.4.11.1+BER/opam
@@ -14,7 +14,7 @@ setenv: CAML_LD_LIBRARY_PATH = "%{lib}%/stublibs"
 build: [
   ["./configure" "-prefix" "%{prefix}%"]
     {os != "openbsd" & os != "freebsd" & os != "macos"}
-  ["./configure" "-prefix" "%{prefix}%" "-cc" "cc" "-aspp" "cc -c"]
+  ["./configure" "-prefix" "%{prefix}%" "CC=cc" "-aspp" "cc -c"]
     {os = "openbsd" | os = "freebsd" | os = "macos"}
   [make "world.opt"]
   [make "-C" "ber-metaocaml-111" "all"]

